### PR TITLE
fix: update code because of landing-zone-vpc v4 release

### PR DESCRIPTION
**Pull request by maheshwarishikha**
### Description

Update the code and variables to accommodate ACL related changes in terraform-landing-zone-vpc v4 release

#### No release required

- [x] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code

#### Release required

- [x] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [ ] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [ ] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content

Replace this text with information that users need to know about the bug fixes, features, and breaking changes. This information helps the merger write the commit message that is published in the release notes for the module.

---

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

File: tests/pr_test.go [hunk 1]
```diff
@@ -19,6 +19,15 @@ const quickstartExampleTerraformDir = "examples/quickstart"
 const roksPatternTerraformDir = "patterns/roks"
 const resourceGroup = "geretain-test-resources"
 
+// Temp: Will be updated once issue https://github.ibm.com/GoldenEye/issues/issues/4302 is fixed
```
Comment (ocofaigh) [inline tests/pr_test.go:22]: lets be clear here - the `atracker_target` ignore is being tracked in https://github.ibm.com/GoldenEye/issues/issues/4302

The ACL ignores can be removed once we merge this PR

Comment (maheshwarishikha) [reply tests/pr_test.go:22]: ok, sure

**Review by rajatagarwal-ibm — APPROVED** [review]

**Comment by ocofaigh** [comment]
Hmm, the test failed with:
```
TestRunUpgradeQuickstartExample 2023-03-30T12:08:05Z retry.go:99: Returning due to fatal error: FatalError{Underlying: error while running command: exit status 1; ╷
│ Error: InvalidArgument: The location specified in the KMS key CRN does not exist, or is not supported.
│ 	status code: 400, request id: 857ad8f4-0ea9-4215-9d53-469e06b4f11c, host id: 
│ 
│   with module.landing_zone.module.landing_zone.ibm_cos_bucket.buckets["management-bucket"],
│   on ../../cos.tf line 74, in resource "ibm_cos_bucket" "buckets":
│   74: resource "ibm_cos_bucket" "buckets" {
│ 
╵
╷
│ Error: InvalidArgument: The location specified in the KMS key CRN does not exist, or is not supported.
│ 	status code: 400, request id: 7a6e288e-3b07-4755-b704-f2d7a17ce305, host id: 
│ 
│   with module.landing_zone.module.landing_zone.ibm_cos_bucket.buckets["workload-bucket"],
│   on ../../cos.tf line 74, in resource "ibm_cos_bucket" "buckets":
│   74: resource "ibm_cos_bucket" "buckets" {
│ 
╵}
```

**Comment by ocofaigh** [comment]
@maheshwarishikha Now pipeline is failing with `Your Cloud Object Storage (COS) bucket is missing the Identity Authorization Management (IAM) authorization to allow the flow logs service to write flow logs to your bucket. (Error response code returned from COS Config call: 403)`

**Comment by maheshwarishikha** [comment]
> Hmm, the test failed with:
> 
> ```
> TestRunUpgradeQuickstartExample 2023-03-30T12:08:05Z retry.go:99: Returning due to fatal error: FatalError{Underlying: error while running command: exit status 1; ╷
> │ Error: InvalidArgument: The location specified in the KMS key CRN does not exist, or is not supported.
> │ 	status code: 400, request id: 857ad8f4-0ea9-4215-9d53-469e06b4f11c, host id: 
> │ 
> │   with module.landing_zone.module.landing_zone.ibm_cos_bucket.buckets["management-bucket"],
> │   on ../../cos.tf line 74, in resource "ibm_cos_bucket" "buckets":
> │   74: resource "ibm_cos_bucket" "buckets" {
> │ 
> ╵
> ╷
> │ Error: InvalidArgument: The location specified in the KMS key CRN does not exist, or is not supported.
> │ 	status code: 400, request id: 7a6e288e-3b07-4755-b704-f2d7a17ce305, host id: 
> │ 
> │   with module.landing_zone.module.landing_zone.ibm_cos_bucket.buckets["workload-bucket"],
> │   on ../../cos.tf line 74, in resource "ibm_cos_bucket" "buckets":
> │   74: resource "ibm_cos_bucket" "buckets" {
> │ 
> ╵}
> ```

I have seen this error..re-ran helped.

**Comment by maheshwarishikha** [comment]
> @maheshwarishikha Now pipeline is failing with `Your Cloud Object Storage (COS) bucket is missing the Identity Authorization Management (IAM) authorization to allow the flow logs service to write flow logs to your bucket. (Error response code returned from COS Config call: 403)`

Need to re-run and check.

**Comment by imprateeksh** [comment]
Now the Failure seen in pipeline is because of gateway limit : 
image
Tried executing all the tests in local env, all tests are passing , attached the logs here :
[TestRunNoComputeExample.log](https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone/files/11119389/TestRunNoComputeExample.log)
[TestRunQuickstartExample.log](https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone/files/11119392/TestRunQuickstartExample.log)

Re-running the pipeline to see if the same error is not seen.

**Comment by ocofaigh** [comment]
I have done some transit gateway cleanup on our account, so we should not hit the quota issue on this pipeline run, however I am concerned about the authorization issues. Are we missing authorization policies or something here?

**Comment by imprateeksh** [comment]
> I have done some transit gateway cleanup on our account, so we should not hit the quota issue on this pipeline run, however I am concerned about the authorization issues. Are we missing authorization policies or something here?

@ocofaigh - I am checking it, I am not sure currently but the pipeline failed again , [Internal Error](https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone/actions/runs/4573075179/jobs/8073062905?pr=315#step:7:25121) is seen for `management` and `atracker` buckets created using resource `ibm_cos_bucket`

**Comment by imprateeksh** [comment]
This looked like an intermittent issue as the tests successfully passed now. 
@ocofaigh - Could you please review and approve this PR?

**Pull request merged**